src/internal/operations/backup_integration_test.go

```diff
@@ -191,67 +206,210 @@ func checkMetadataFilesExist(
-	files []string,
+	filesByCat map[path.CategoryType][]string,
 ) {
 	//revive:enable:context-as-argument
-	bup := &backup.Backup{}
+	for category, files := range filesByCat {
+		t.Run(category.String(), func(t *testing.T) {
+			bup := &backup.Backup{}
 
-	err := ms.Get(ctx, model.BackupSchema, backupID, bup)
-	if !assert.NoError(t, err) {
-		return
+			err := ms.Get(ctx, model.BackupSchema, backupID, bup)
+			if !assert.NoError(t, err) {
+				return
+			}
+
+			paths := []path.Path{}
+			pathsByRef := map[string][]string{}
+
+			for _, fName := range files {
+				p, err := path.Builder{}.
+					Append(fName).
+					ToServiceCategoryMetadataPath(tenant, user, service, category, true)
+				if !assert.NoError(t, err, "bad metadata path") {
+					continue
+				}
+
+				dir, err := p.Dir()
+				if !assert.NoError(t, err, "parent path") {
+					continue
+				}
+
+				paths = append(paths, p)
+				pathsByRef[dir.ShortRef()] = append(pathsByRef[dir.ShortRef()], fName)
+			}
+
+			cols, err := kw.RestoreMultipleItems(ctx, bup.SnapshotID, paths, nil)
+			assert.NoError(t, err)
+
+			for _, col := range cols {
+				itemNames := []string{}
+
+				for item := range col.Items() {
+					assert.Implements(t, (*data.StreamSize)(nil), item)
+
+					s := item.(data.StreamSize)
+					assert.Greaterf(
+						t,
+						s.Size(),
+						int64(0),
+						"empty metadata file: %s/%s",
+						col.FullPath(),
+						item.UUID(),
+					)
+
+					itemNames = append(itemNames, item.UUID())
+				}
+
+				assert.ElementsMatchf(
+					t,
+					pathsByRef[col.FullPath().ShortRef()],
+					itemNames,
+					"collection %s missing expected files",
+					col.FullPath(),
+				)
+			}
+		})
 	}
+}
 
-	paths := []path.Path{}
-	pathsByRef := map[string][]string{}
+// ---------------------------------------------------------------------------
+// Incremental Item Generators
+// TODO: this is ripped from factory.go, which is ripped from other tests.
+// At this point, three variation of the sameish code in three locations
+// feels like something we can clean up.  But, it's not a strong need, so
+// this gets to stay for now.
+// ---------------------------------------------------------------------------
 
-	for _, fName := range files {
-		p, err := path.Builder{}.
-			Append(fName).
-			ToServiceCategoryMetadataPath(tenant, user, service, category, true)
-		if !assert.NoError(t, err, "bad metadata path") {
-			continue
-		}
+// the params here are what generateContainerOfItems passes into the func.
+// the callback provider can use them, or not, as wanted.
+type dataBuilderFunc func(id, now, subject, body string) []byte
 
-		dir, err := p.Dir()
-		if !assert.NoError(t, err, "parent path") {
-			continue
-		}
+//revive:disable:context-as-argument
+func generateContainerOfItems(
+	t *testing.T,
+	ctx context.Context,
+	gc *connector.GraphConnector,
+	service path.ServiceType,
+	cat path.CategoryType,
+	sel selectors.Selector,
+	tenantID, userID, destFldr string,
+	howMany int,
```

Comment:
```suggestion
     total int,
```
The variable name would need to be updated within the function.